packages/jest-types/__typetests__/each.test.ts

```diff
@@ -217,8 +217,8 @@ expectType<void>(
     ${'a'} | ${true}
     ${'b'} | ${false}
   `('some test', ({item, expected}) => {
-    expectType<unknown>(item);
-    expectType<unknown>(expected);
+    expectType<string | boolean>(item);
```

Comment:
Hm.. to be honest I find this behaviour confusing. `item` cannot be `boolean`. Again, just an opinion, but isn't it better to explicitly tell that type cannot be inferred instead of providing misleading / incorrect type?

Reply:
Hmm, I see what you mean. Not sure if there is a standard & this might ultimately be opinion.

To clarify my reasoning, there are 2 advantages to using a union imo:

1. Less need to cast. When 1 input can actually be of different types (this might not happen often I recon, except maybe for generic functions):
```
test.each`
  a       | expected
  ${0}    | ${'0'}
  ${true} | ${'true'}
  ${'a'}  | ${'a'}
`('strange stringify', ({a, expected}) => {
  expect(toString(a)).toBe(expected); // no need to cast `a` here with union
});
```

2. Identify abusive casting: you can cast `unknown` to anything, but not the union.
(the following example is strange, but something similar could happen during refactoring or when a test is updated)
```
test.each`
  a    | b    | expected
  ${1} | ${2} | ${'three'}
  ${3} | ${4} | ${'seven'}
`('strange stringify', ({a, expected}) => {
  const shouldError = expected as any[]
  // With `unknown`: no error 😬
  // With union: Conversion of type 'string | number' to type 'any[]' may be a mistake because neither type sufficiently overlaps with the other.
});
```

wdyt?

Reply:
Thanks for the proposal, but I don’t see it as improvement. Wrong types are wrong types (;

Reply:
I think it's an improvement to say "this can be a union of these types" over `unknown`, even if the union is too broad.